app/main.py

```diff
@@ -1,2 +1,89 @@
+from typing import Any, Hashable, Iterator
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 0.66
+        self.table: list = [None] * self.capacity
+        self.length = 0
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.length / self.capacity > self.load_factor:
+            self._resize()
+
+        index = hash(key) % self.capacity
+
+        while self.table[index]:
+            node_key, _ = self.table[index]
+            if node_key == key:
+                self.table[index] = (key, value)
+                return
+            index = (index + 1) % self.capacity
+
+        self.table[index] = (key, value)
+        self.length += 1
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = hash(key) % self.capacity
+
+        while self.table[index]:
+            node_key, node_value = self.table[index]
+            if node_key == key:
+                return node_value
+            index = (index + 1) % self.capacity
+
+        raise KeyError(f"Key '{key}' not found.")
+
+    def __len__(self) -> int:
+        return self.length
+
+    def _resize(self) -> None:
+        self.capacity *= 2
+        storage = [*self.table]
+        self.table = [None] * self.capacity
+
+        for node in storage:
+            if node:
+                key, value = node
+                new_index = hash(key) % self.capacity
+                while self.table[new_index]:
+                    new_index = (new_index + 1) % self.capacity
+                self.table[new_index] = (key, value)
+
+    def clear(self) -> None:
+        self.table = [None] * self.capacity
+        self.length = 0
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = hash(key) % self.capacity
+
+        while self.table[index]:
+            node_key, _ = self.table[index]
+            if node_key == key:
+                self.table[index] = None
```

Comment:
The `__delitem__` method sets the table entry to `None` but does not handle rehashing of subsequent items in the cluster. This can break the probing sequence and cause items to become unreachable. Consider implementing a rehashing mechanism to maintain the integrity of the hash table after deletion.